modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java

```diff
@@ -4150,4 +4149,49 @@ public void testUnalias() {
         Assert.assertEquals(codegenParameter.defaultValue, "1971-12-19T03:39:57-08:00");
         Assert.assertEquals(codegenParameter.getSchema(), null);
     }
+
+    @Test
+    public void testArraySchemaWithIneffectiveConstraints(){
+
+        final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/issue6491.yaml");
+        final DefaultCodegen codegen = new DefaultCodegen();
+        codegen.setOpenAPI(openAPI);
+
+        String modelName = "ArrayWithIneffectiveValidations";
+        Schema sc = openAPI.getComponents().getSchemas().get(modelName);
+        CodegenModel cm = codegen.fromModel(modelName, sc);
```

Comment:
- Please assert that warnings are being logged for validations for your test cases